src/repr/src/adt/interval.rs

```diff
@@ -55,10 +55,10 @@ impl Interval {
     /// Constructs a new `Interval` with the specified units of time.
     ///
     /// `nanos` in excess of `999_999_999` are carried over into seconds.
-    pub fn new(months: i32, seconds: i64, nanos: i64) -> Result<Interval, anyhow::Error> {
+    pub fn new(months: i32, seconds: i64, nanos: i128) -> Result<Interval, anyhow::Error> {
```

Comment:
Just removed `as` from as many places as possible. There were two places were I needed to use `as` because it was converting from an `f64` to `i128`, but there are checks right before the conversion. In a couple of places I had to use `from` just because there was some arithmetic happening in the function call.